crates/autopilot/src/run_loop.rs

```diff
@@ -699,10 +734,45 @@ impl RunLoop {
             "settlement transaction await reached deadline"
         )))
     }
+
+    /// Removes orders that are currently being settled to avoid solvers trying
+    /// to fill an order a second time.
+    async fn remove_in_flight_orders(&self, mut auction: domain::Auction) -> domain::Auction {
+        let Some(in_flight) = &*self.in_flight_orders.lock().await else {
+            return auction;
+        };
+
+        let transaction = self.eth.transaction(in_flight.tx_hash.into()).await;
```

Comment:
Since the `/settle` task knows which orders it tries to settle I think we no longer need to have the `tx_hash` (probably an error from the old code).
The cache could be a simple `HashSet<OrderUid>`. Before spawning the `/settle` task we add all the to be settled orders and after the task finished we remove them from the cache again.